### DISCUSSION TITLE
关于后续开发方向的一点想法

### DISCUSSION THREAD

**Discussion by 1m188** [MLAcookie/MAA1999WPF#11] [Ideas]
### 关于后续开发方向的一点想法

刚刚注意到 https://github.com/MLAcookie/M9AWPF/pull/9#issue-2152772597 这个PR已经被合并了，在这里简单提一点关于后续的开发方向的一点想法，也欢迎大家讨论完善。

1. 基本的维护和完善。
1.1. 注释。注意到很多地方的注释都不够完善，在开发的时候往往需要仔细阅读具体实现才能明白一个函数或者属性的功能；同时为了更好的划分文件和模块之间的功能，也建议在每个文件顶上进行注释说明该文件的功能和内容等。
1.2. 命名。应遵循C#推荐的命名方式，函数，属性，类等均采用大驼峰命名法，而变量，私有/公有变量应采用小驼峰命名法等。
1.3. 格式化。我听说.net可以通过.editconfig文件来控制格式化，或许可以加入这个文件以规范开发的格式。
1.4. 精简和合并。感觉有点地方有些设计过度了，比如说ConfigManager.cs和MAAConfigObject.cs这两个文件，或许可以把MAAConfigObject.cs中的Interface部分提出来单独作为一个文件用于读取MAA的所有接口配置，而MAAConfigObject和Manager可以合并，由一个非静态的类（非静态的类为后面如果要实现多实例的话或许有利）来实现从本地config json读取配置，并对外提供获取和修改配置的接口，以及再重新写入本地文件实现持久化等功能的类，然后由ViewModel拿到这个类的一个实例，再做一层包装提供给View操作功能。当然这里只是举个例子，别的地方可能也有类似的情况。
1.5. 其他。比如说现在按下Start按钮之后仍然会出现黑框框，并且任务流程执行完之后不会自动关闭，如果能针对做些体验上的优化或许会更好等等之类的。

2. M9A的版本管理。之前的一个PR https://github.com/MLAcookie/M9AWPF/pull/6#issue-2133671531 里， @KaronGH 提醒我才发现用的M9A的版本已经落后了，之前针对旧版本的修改必须进一步修改。感觉目前的对于M9A的集成方式不太好统一所有开发者的版本，对于构建发布项目的时候或许也会产生一些不方便的地方。我个人的一点想法是，或许可以在解决方案下加入一个项目，然后M9AWPF构建的时候依赖该项目，该项目没有内容，只是在构建时会检查本地是否存在对应的M9A目录，如果不存在的话则会通过指定的链接进行下载，然后在M9AWPF构建的时候M9A也会一并复制到构建目录下。不知道MS Build有没有对应的功能可以实现类似的功能。。。。。。这样子做的好处是可以通过几行文本的配置文件控制M9A的版本，而当类似上次那样M9A对外接口的重大修改的时候也可以快速便捷的跟上版本。

3. 定时器功能。感觉这个功能还蛮重要的，因为可以7x24小时不停的刷233。。。

4. 多实例功能。在针对多个模拟器多个游戏的情况下或许有用。

5. 更新模块。能够及时跟上最新的活动。

6. 注意到maa文档里有代码集成的二次开发方法，或许可以只使用M9A的resource和interface而直接代码集成maa以加强对功能的定制能力？。。。

目前大概就是这么多，我对于这方面也不太熟，欢迎各位大佬讨论完善

**Comment by MLAcookie** [comment]
我也按一条条回复下
1. 
    1.1. 现在基本功能也稳定了，注释这块也是我打算做的
    1.2. 命名这块是我的问题，头一次做多人开发，命名这块就按微软的来吧
    1.3. 这个功能是有的，具体的选项我再开个讨论吧
    1.4. 设计过度的问题我觉得是暂时的问题，目前不是很着急
    1.5. 这个还是缺少测试的问题，慢慢完善吧
2. 构建的问题是我之前经验不足，现在就在研究这个问题
3. 定时器功能也打算做的
4. 很好的想法(什么人会养好几个1999号啊)
5. 这个也是打算做的
6. 这个感觉有点远了，不过我也看到MBA项目是用C#集成的，之后可以研究下
